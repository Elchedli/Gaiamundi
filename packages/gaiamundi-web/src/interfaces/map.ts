import { ApiSingleContentType } from './api';
import { User } from './user';

export interface GeoMapAttributes {
  name: string;
  owner?: ApiSingleContentType<User>;
  yearValidity?: number;
  source?: string;
  license?: string;
  geojson: JSON;
}
