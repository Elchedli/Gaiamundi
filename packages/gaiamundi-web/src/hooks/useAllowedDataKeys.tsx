import { useMemo } from 'react';
import { ChartType, RawDatumType } from '../interfaces/chart';
import { useChartConfig } from './useChartConfig';
import { useData } from './useData';

export const ALLOWED_DOMAIN_TYPES_BY_CHART: {
  [key in ChartType]: Array<RawDatumType>;
} = {
  column: ['string', 'number'],
  bar: ['string', 'number'],
  scatter: ['number'],
  line: ['number'],
  lineColumn: ['string', 'number'],
  pie: ['number'],
  area: ['number'],
  bubble: ['number'],
};

export const useAllowedDataKeys = () => {
  const { chart: chartConfig } = useChartConfig();
  const { dataKeys } = useData();
  const allowedTypes = ALLOWED_DOMAIN_TYPES_BY_CHART[chartConfig.type];

  const domainKeyOptions = useMemo(
    () =>
      Object.entries(dataKeys)
        .filter(([, type]) => {
          return allowedTypes.indexOf(type) !== -1;
        })
        .map(([key]) => ({
          value: key,
          label: key,
        })),
    [allowedTypes, dataKeys]
  );

  return { domainKeyOptions };
};
