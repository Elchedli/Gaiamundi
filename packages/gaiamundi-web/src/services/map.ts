import { Map } from '../interfaces/map';
import { strapi } from './init';

export const getMap = async (query?: object | number) => {
  return await strapi.get('maps', query);
};

export const createMap = async (data: Map) => {
  return await strapi.create('maps', data);
};

export const updateMap = async (id: number, data: Partial<Map>) => {
  return await strapi.update('maps', id, data);
};

export const deleteMap = async (id: number) => {
  return await strapi.delete('maps', id);
};
