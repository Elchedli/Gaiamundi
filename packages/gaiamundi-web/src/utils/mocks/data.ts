import { ApiCollection, ApiData } from 'interfaces/api';
import { DataFragment } from 'interfaces/data-fragment';
import { Dataset } from 'interfaces/dataset';
import { UploadedFile } from 'interfaces/file';
import { GeoMap } from 'interfaces/geo-map';
import { Indicator } from 'interfaces/indicator';
import { PageCarto } from 'interfaces/page-carto';
import { Tag } from 'interfaces/tag';
import { User } from 'interfaces/user';
import mockMapJson from './map.json';

export const mockMapPath = mockMapJson;

export const mockFileUpload = {
  id: 184,
  name: 'exampleParsing2.csv',
  alternativeText: null,
  caption: null,
  width: null,
  height: null,
  formats: null,
  hash: 'example_Parsing2_ce8756a044',
  ext: '.csv',
  mime: 'text/csv',
  size: 0.05,
  url: '/uploads/example_Parsing2_ce8756a044.csv',
  previewUrl: null,
  provider: 'local',
  provider_metadata: null,
  createdAt: '2023-03-16T18:20:09.476Z',
  updatedAt: '2023-03-16T18:20:09.476Z',
};

export const mockUser: User = {
  id: 1,
  username: 'testuser',
  password: 'test',
  email: 'test@test.com',
  provider: 'local',
  confirmed: true,
  blocked: false,
  created_at: '2022-02-07T00:00:00.000Z',
  updated_at: '2022-02-07T00:00:00.000Z',
};

export const mockGeoJson: ApiData<UploadedFile> = {
  id: 1,
  name: 'test.geojson',
  hash: 'abc123',
  provider: 'provider123',
  ext: '.geojson',
  mime: 'application/json',
  size: 1234,
  url: '/api/v1/geo-maps/id/thumbnail',
  formats: {
    thumbnail: {
      extension: 'ext',
      url: 'happyUrl',
      size: 360,
      hash: 'hashbar',
      mime: 'mimeExample',
    },
  },
  created_at: '2022-02-07T00:00:00.000Z',
  updated_at: '2022-02-07T00:00:00.000Z',
};

export const mockCover: ApiData<UploadedFile> = {
  id: 3,
  name: 'image.png',
  hash: 'hashImage',
  provider: 'provider123',
  ext: '.png',
  mime: 'application/image',
  size: 1234,
  url: '/api/v1/geo-maps/id/thumbnailImage',
  formats: {
    small: {
      extension: 'ext',
      url: '/api/v1/geo-maps/id/thumbnailImage?720',
      size: 360,
      hash: 'hash720',
      mime: 'mimeExample',
    },
    thumbnail: {
      extension: 'ext',
      url: '/api/v1/geo-maps/id/thumbnailImage?720',
      size: 360,
      hash: 'hash720',
      mime: 'mimeExample',
    },
  },
  created_at: '2022-02-07T00:00:00.000Z',
  updated_at: '2022-02-07T00:00:00.000Z',
};

export const mockGeoMap: ApiData<GeoMap> = {
  id: 1,
  name: 'Test Map',
  yearValidity: 2022,
  mesh: 'Test Maille',
  source: 'Test Source',
  license: 'Test License',
  properties: [
    {
      name: 'code',
      sample: '39',
      isGeoCode: true,
    },
    {
      name: 'nom',
      sample: 'Jura',
      isGeoCode: false,
    },
  ],
  owner: mockUser,
  geoJSON: mockGeoJson,
  created_at: '2022-02-07T00:00:00.000Z',
  updated_at: '2022-02-07T00:00:00.000Z',
};

export const mockTags: ApiData<Tag>[] = [
  {
    id: 1,
    name: 'Tag A',
    count: 5,
    created_at: '2022-02-07T00:00:00.000Z',
    updated_at: '2022-02-07T00:00:00.000Z',
  },
  {
    id: 2,
    name: 'Tag B',
    count: 3,
    created_at: '2022-02-07T00:00:00.000Z',
    updated_at: '2022-02-07T00:00:00.000Z',
  },
  {
    id: 3,
    name: 'Tag C',
    count: 2,
    created_at: '2022-02-07T00:00:00.000Z',
    updated_at: '2022-02-07T00:00:00.000Z',
  },
];

export const mockDataset: ApiData<Dataset> = {
  id: 122,
  name: 'firstDataset',
  origin: 3,
  isPublic: true,
  owner: mockUser,
  csv: {
    id: 5,
    name: 'file.csv',
    hash: 'hashCSV',
    provider: 'provider123',
    ext: '.csv',
    mime: 'application/csv',
    size: 1234,
    url: '/api/v1/Uploads/id/csvFile.csv',
    formats: {
      thumbnail: {
        extension: 'ext',
        url: '/api/v1/Uploads/id/csvFile.csv?720',
        size: 360,
        hash: 'hash720',
        mime: 'mimeExample',
      },
    },
    created_at: '2022-02-07T00:00:00.000Z',
    updated_at: '2022-02-07T00:00:00.000Z',
  },
  created_at: '2022-02-07T00:00:00.000Z',
  updated_at: '2022-02-07T00:00:00.000Z',
};

export const mockDataFragments: ApiData<DataFragment>[] = [
  {
    id: 1,
    name: 'FirstFragment',
    columns: [
      {
        name: 'firstColumn',
        sample: 5,
        source: 'source1',
        validity: '2021',
        isGeoCode: true,
      },
      {
        name: 'secondColumn',
        sample: 10.3,
        source: 'source2',
        validity: '2022',
        isGeoCode: false,
      },
      {
        name: 'thirdColumn',
        sample: 0.25,
        source: 'source3',
        validity: '2023',
        isGeoCode: false,
      },
      {
        name: 'fourthColumn',
        sample: 1000,
        source: 'source4',
        validity: '2024',
        isGeoCode: false,
      },
    ],
    dataset: mockDataset,
    created_at: '2022-02-07T00:00:00.000Z',
    updated_at: '2022-02-07T00:00:00.000Z',
  },
];

export const indicatorsMockData: ApiData<Indicator>[] = [
  {
    id: 4,
    name: 'mockIndicator1',
    description: 'This is a mock indicator',
    source: 'mock',
    validity: 2,
    created_at: '2023-03-14T10:06:41.979Z',
    updated_at: '2023-03-14T10:06:41.979Z',
    equation: 'A+2',
    variables: [],
  },
  {
    id: 5,
    name: 'mockIndicator2',
    description: 'This is a mock indicator',
    source: 'mock',
    validity: 4,
    variables: [],
    created_at: '2023-03-14T10:12:37.285Z',
    updated_at: '2023-03-14T10:12:37.285Z',
    equation: 'A*2',
  },
];

export const mockPageCartoData: ApiData<PageCarto> = {
  id: 1,
  name: 'Page Carto 1',
  owner: mockUser,
  tags: mockTags,
  map: mockGeoMap,
  cover: mockCover,
  indicators: indicatorsMockData,
  charts: [],
  html: '<p>Test HTML</p>',
  data_fragments: mockDataFragments,
  created_at: '2022-02-07T00:00:00.000Z',
  updated_at: '2022-02-07T00:00:00.000Z',
};

export const mockGeoMapData: ApiData<GeoMap> = {
  id: 1,
  name: 'Test Map',
  yearValidity: 2022,
  mesh: 'Test Maille',
  source: 'Test Source',
  license: 'Test License',
  properties: [
    {
      name: 'nom',
      sample: 'Taponas',
      isGeoCode: true,
    },
  ],
  owner: mockUser,
  geoJSON: {
    id: 1,
    name: 'test.geojson',
    hash: 'abc123',
    provider: 'provider123',
    ext: '.geojson',
    mime: 'application/json',
    size: 1234,
    url: 'https://example.com/api/v1/geo-maps/id/thumbnail',
    formats: {
      '720': {
        extension: 'ext',
        url: 'happyUrl',
        size: 360,
        hash: 'hashbar',
        mime: 'mimeExample',
      },
    },
    created_at: '2022-02-07T00:00:00.000Z',
    updated_at: '2022-02-07T00:00:00.000Z',
  },
  created_at: '2022-02-07T00:00:00.000Z',
  updated_at: '2022-02-07T00:00:00.000Z',
};

export const mockGeoMapApiCollection: ApiCollection<GeoMap> = {
  data: Array.from({ length: 3 }, (_, index) => ({
    ...mockGeoMapData,
    id: index + 1,
    name: `Test Map ${index + 1}`,
  })),
  meta: {
    pagination: {
      pageCount: 1,
      total: 3,
    },
  },
};
