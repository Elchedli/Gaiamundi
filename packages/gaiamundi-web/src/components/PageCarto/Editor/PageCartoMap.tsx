import { CameraIcon, MinusIcon, PlusIcon } from '@heroicons/react/24/solid';
import 'eazychart-css';
import { MapChart, ResponsiveChartContainer } from 'eazychart-react';
import { Feature } from 'interfaces/geojson';
import panzoom, { PanZoom } from 'panzoom';
import { FC, useCallback, useEffect, useMemo, useRef } from 'react';
import { useQuery } from 'react-query';

import rewind from '@turf/rewind';
import { Alert } from 'components/Alert/Alert';
import { ApiErrorAlert } from 'components/Alert/ApiErrorMessage';
import { Button } from 'components/Button/Button';
import ButtonGroup from 'components/Button/ButtonGroup';
import { LoadingMessage } from 'components/Loader/LoadingMessage';
import { usePageCarto } from 'hooks/usePageCarto';
import { ApiData, ApiError } from 'interfaces/api';
import { UploadedFile } from 'interfaces/file';
import { getGeoJson } from 'services/geo-map';
import { uploadCover } from 'services/page-carto';
import { extractScreenshotBySvg } from 'utils/exportChartAsPng';

export const PageCartoMap: FC = () => {
  const elementRef = useRef<SVGSVGElement | null>(null);
  const panzoomRef = useRef<PanZoom | null>(null);
  const { map, pageCartoId } = usePageCarto();
  const geoJson = map?.geoJSON;
  const { data, isError, isLoading, isIdle, error } = useQuery({
    queryKey: ['geoJSON', geoJson?.id],
    queryFn: async () => getGeoJson(geoJson as ApiData<UploadedFile>),
    // The query will not execute until the userId exists
    enabled: !!geoJson,
  });

  const geoCode = useMemo(() => {
    const geoCodeProperty = map?.properties.find((property) => {
      return property.isGeoCode === true;
    });
    return geoCodeProperty ? geoCodeProperty.name : 'admin';
  }, [map]);

  //Whenever the map values change the saveScreenshot is triggered.
  //this still unsure about useeffect and map.geoJSON updating.
  useEffect(() => {
    saveScreenshot();
  }, [map?.geoJSON]);

  // Set up panzoom on mount, and dispose on unmount
  const panZoomCallback = useCallback((element: HTMLDivElement | null) => {
    if (element) {
      elementRef.current = element.querySelector('svg');
      if (elementRef.current) {
        panzoomRef.current = panzoom(elementRef.current, {
          zoomSpeed: 0.1,
          minZoom: 0.25,
          maxZoom: 4,
          smoothScroll: true,
          bounds: true,
          boundsPadding: 0.4,
        });
      }
    }
  }, []);

  const zoomIn = () => {
    if (elementRef.current) {
      const rect = elementRef.current.getBBox();
      const cx = rect.x + rect.width / 2;
      const cy = rect.y + rect.height / 2;
      panzoomRef.current?.smoothZoom(cx, cy, 1.25);
    }
  };

  const zoomOut = () => {
    if (elementRef.current) {
      const rect = elementRef.current.getBBox();
      const cx = rect.x + rect.width / 2;
      const cy = rect.y + rect.height / 2;
      panzoomRef.current?.smoothZoom(cx, cy, 0.75);
    }
  };

  const saveScreenshot = async () => {
    if (elementRef.current) {
      //clone the svg dom so i can reset the place to it's initial position and size for the perfect picture
      const svgData = elementRef.current.cloneNode(true) as SVGSVGElement;
      // svg container width and height
      svgData.setAttribute(
        'width',
        elementRef.current.width.baseVal.value.toString()
      );
      svgData.setAttribute(
        'height',
        elementRef.current.height.baseVal.value.toString()
      );
      //this line resets the size and place to the default placement
      svgData.style.transform = 'none';
      //this function is the one the svg and make it into an image.
      const mapScreenshot = await extractScreenshotBySvg(svgData);
      //the new image is sended to the api and change it for the specific pageCartos
      uploadCover(
        mapScreenshot,
        pageCartoId.toString(),
        map?.name + '_screenshot.png'
      );
    }
  };

  const geoJsonData = useMemo(() => {
    return data
      ? {
          ...data,
          features: data.features.map((feature: Feature) => {
            return rewind(feature as any, { reverse: true });
          }),
        }
      : {};
  }, [data]);

  if (isLoading || isIdle) {
    return (
      <LoadingMessage
        label={'Chargement de la carte ...'}
        data-testid="loading-message"
      />
    );
  }

  if (isError) {
    return <ApiErrorAlert error={error as ApiError} />;
  }

  if (!data) {
    return <Alert type="info">Impossible de charger la carte.</Alert>;
  }

  return (
    <div className="w-full h-full relative">
      <ButtonGroup pill={true} className="m-4 absolute z-50">
        <Button icon={PlusIcon} onClick={zoomIn} />
        <Button icon={MinusIcon} onClick={zoomOut} />
        <Button icon={CameraIcon} onClick={saveScreenshot} />
      </ButtonGroup>
      <div
        className="w-full h-full overflow-hidden"
        ref={panZoomCallback}
        data-testid={'map-chart'}
      >
        <ResponsiveChartContainer>
          <MapChart
            map={{
              geoDomainKey: geoCode,
              valueDomainKey: 'value',
              projectionType: 'geoMercator',
              stroke: 'black',
              fill: 'white',
            }}
            padding={{ top: 0, right: 50, bottom: 150, left: 50 }}
            colors={['white', 'pink', 'red']}
            geoJson={geoJsonData}
            data={data.features.map((feature: Feature, idx: number) => {
              return {
                [geoCode]: feature.properties?.[geoCode],
                value: idx,
              };
            })}
          />
        </ResponsiveChartContainer>
      </div>
    </div>
  );
};
