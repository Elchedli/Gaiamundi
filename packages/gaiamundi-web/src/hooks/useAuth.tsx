import React from 'react';
import {
  QueryObserverResult,
  RefetchOptions,
  UseMutateFunction,
  useQuery,
  useQueryClient,
} from 'react-query';

import { Alert } from 'components/Alert/Alert';
import { LoadingMessage } from 'components/Loader/LoadingMessage';
import { User } from 'interfaces/user';
import { strapi } from '../services/strapi';
import { useLocalStorageState } from './useLocalStorageState';

// const updateAccount = async (id: number, userData: UserSignUpFields) => {
//   return (await strapi.update(ContentType.USERS, id, userData)) as User;
// };

export interface AuthContextValue {
  user: User | undefined;
  isAuthenticated: boolean;
  setUser: (data: User | undefined) => void;
  authenticate: (user: User, jwt: string) => void;
  logout: UseMutateFunction;
  refetchUser: (
    options?: RefetchOptions | undefined
  ) => Promise<QueryObserverResult<User, Error>>;
  error: Error | null;
}

const AuthContext = React.createContext<AuthContextValue | null>(null);
AuthContext.displayName = 'AuthContext';

export interface AuthProviderProps {
  children: React.ReactNode;
}

export const AuthProvider = ({ children }: AuthProviderProps): JSX.Element => {
  const { value: jwtToken, persistValue: persistJwtToken } =
    useLocalStorageState('token');
  const queryClient = useQueryClient();
  const key = 'auth-user';

  const {
    data: user,
    error,
    isLoading,
    refetch,
  } = useQuery<User, Error>({
    queryKey: key,
    queryFn: async () => {
      return await strapi.currentUser(jwtToken);
    },
    enabled: !!jwtToken,
  });

  const setUser = (data: User | undefined) =>
    queryClient.setQueryData(key, data);

  const authenticate = (user: User, jwt: string) => {
    setUser(user);
    persistJwtToken(jwt);
  };

  const logout = () => {
    persistJwtToken('');
    queryClient.clear();
  };

  if (isLoading) {
    return <LoadingMessage />;
  }

  return (
    <AuthContext.Provider
      value={{
        user,
        isAuthenticated: !!jwtToken,
        error,
        setUser,
        authenticate,
        refetchUser: refetch,
        logout,
      }}
    >
      {error && <Alert>{error.message}</Alert>}
      {children}
    </AuthContext.Provider>
  );
};

export const useAuth = () => {
  const context = React.useContext(AuthContext);
  if (!context) {
    throw new Error(`useAuth must be used within an AuthProvider`);
  }
  return context;
};
