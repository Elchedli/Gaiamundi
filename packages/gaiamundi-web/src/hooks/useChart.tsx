import { CHART_TYPES } from 'interfaces/chart';
import { useMemo } from 'react';
import { useQuery } from 'react-query';
import { getChartById } from 'services/chart';

export const useChart = (chartId: number) => {
  const query = useQuery({
    queryKey: ['chart', chartId],
    queryFn: async () => await getChartById(chartId),
    enabled: !!chartId,
  });
  const chart = query.data?.data;
  const ChartComponent = useMemo(() => {
    return chart?.type
      ? CHART_TYPES[chart.type].ChartComponent
      : CHART_TYPES.column.ChartComponent;
  }, [chart]);

  return {
    ...query,
    ChartComponent,
  };
};
