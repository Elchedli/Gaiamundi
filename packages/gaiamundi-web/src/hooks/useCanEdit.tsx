import { useLocation } from 'react-router-dom';
import { useAuth } from './useAuth';

export const useCanEdit = () => {
  const { isAuthenticated } = useAuth();
  const location = useLocation();
  const isOnEditorPage = location.pathname.endsWith('edit');
  return isAuthenticated && isOnEditorPage;
};
