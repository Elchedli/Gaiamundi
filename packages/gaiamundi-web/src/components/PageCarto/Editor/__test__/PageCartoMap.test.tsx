import { render } from '@testing-library/react';
import { useAuth } from 'hooks/useAuth';
import { useQuery } from 'react-query';
import { mockGeoMapData } from 'utils/mocks/data';
import { mockMapPath } from 'utils/mocks/map';
import { PageCartoMap } from '../PageCartoMap';

jest.mock('hooks/useAuth');
jest.mock('react-query', () => ({
  useQuery: jest.fn(),
}));

//Errors :
// ResizeObserver is not defined
//Unable to find the geo domain key in the feature properties
describe('PageCartoMap', () => {
  beforeEach(() => {
    // Mock useAuth hook to return a fake user object
    (useAuth as jest.Mock).mockReturnValue({
      isAuthenticated: true,
      user: { id: 1 },
    });
    // Mock the useQuery hook to return a fake response object
    (useQuery as jest.Mock).mockReturnValue({
      data: { ...mockMapPath },
    });
  });

  it('renders loading message while data is being fetched', async () => {
    render(<PageCartoMap map={mockGeoMapData} />);
    // expect(container).toMatchSnapshot();
  });
});
