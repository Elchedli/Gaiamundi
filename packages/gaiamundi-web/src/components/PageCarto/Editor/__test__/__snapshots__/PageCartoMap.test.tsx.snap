// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageCartoMap renders map once geojson data is fetched 1`] = `
<div
  class="w-full h-full overflow-hidden"
  data-testid="map-chart"
>
  <div
    class="ez-responsive-container"
    style="width: 100%; height: 100%;"
  >
    <div
      class="ez-viz"
      style="width: 800px; height: 600px;"
    >
      <div
        class="ez-fragment-tooltip"
      >
        <div
          class="ez-fragment-filter"
        >
          <div
            class="ez-chart"
            tabindex="0"
          >
            <svg
              class="ez-svg"
              style="width: 800px; height: 600px;"
            >
              <g
                transform="translate(50, 0)"
              >
                <g
                  class="ez-fragment-color-scale"
                >
                  <g
                    class="ez-map"
                  >
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                    <path
                      class="ez-map-path"
                      d=""
                      fill="black"
                      stroke="black"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1"
                    />
                  </g>
                </g>
              </g>
            </svg>
          </div>
        </div>
        <div
          class="ez-tooltip"
          style="left: 0px; top: 0px; opacity: 0;"
        />
      </div>
    </div>
  </div>
</div>
`;
